scripts/gpu2ranks_slurm

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Usage: `mpirun gpu2ranks_slurm <my_commands>`
+#
+# This script is used to restrict visible GPUs when using multiple ranks per
+# node with slurm. It can be used with CUDA or HIP.
+
+set -eu
+
+export CUDA_VISIBLE_DEVICES=$SLURM_LOCALID
+export ROCR_VISIBLE_DEVICES=$SLURM_LOCALID
```

Comment:
Might be dangerous. Is it not better to have system specific scripts that take care of GPU "position"?
(I see that for lumi you use a special CPU binding to take care of it what about hohgant?)

Reply:
There are two options for how to do the remapping: 1. Remap GPUs based on rank or 2. Remap "ranks" based on GPUs. For LUMI the scripts now do 2., i.e. available cores are remapped for each rank so that the rank-GPU mapping is the straightforward `{CUDA,ROCR}_VISIBLE_DEVICES=$SLURM_LOCALID`. If we do option 2. on all systems this will will work fine on those systems as well, but I definitely see your point about this being potentially dangerous.  

Speaking of which, I obviously should've done the same for hohgant... If I'm interpreting the output of `nvidia-smi` correctly the mapping is reversed (first numa domain has affinity to the last GPU etc.). I'll do some checks there before updating this again.

Reply:
I added masks for hohgant based on `rocm-smi`/`nvidia-smi` (see comments in `systems.py`). The hohgant topology on `amdgpu` is the same as on lumi. I ran a few tests with `miniapp_cholesky`/`miniapp_triangular_solver` with both the "naive"/"standard" bindings and the "affinity" bindings on `nvgpu` and noticed no difference, but I've left the mapping in place because it should not hurt at least.

Reply:
I have not done anything about the `gpu2ranks_slurm` script. If you prefer I can rename this into two `gpu2ranks_{hohgant,lumi}` scripts?

Reply:
No it is fine. However, please add a note in this script that GPU affinity has to be taken care in systems.py.

Reply:
See https://github.com/eth-cscs/DLA-Future/pull/786/commits/7ffdcdebf62a52b7d7f3639c76145d266026f413. Is that ok?